libs/datasets/timeseries.py

```diff
@@ -98,7 +98,7 @@ def yield_records(self) -> Iterable[dict]:
         # It'd be faster to use self.data.itertuples or find a way to avoid yield_records, but that
         # needs larger changes in code calling this.
         for idx, row in self.data.iterrows():
-            yield row.where(pd.notnull(row), None).to_dict()
+            yield row.loc[row.notna()].to_dict()
```

Comment:
> What was the motivation for removing columns with None values?
> 
> I'm a bit hesitant at the change because it feels like we're loosening the API of what a dataset is returning. It seems like two different statements to say "a specific column is included in the dataset, so there is a missing value for that row" vs "check to see if a specific column has a value regardless if it's in the dataset or not". `data[CommonFields.CASES]` is a much stronger guarantee and gives less flexibility to the code using `data` than `data.get(CommonFields.CASES)`.
> 
> I may totally be overthinking, but just want to double check the motivation for the change and see if there's maybe another easier way to accomplish a similar goal!

I'd like to replace all use of yield_records and friends for getting access to the latest values with access to the dict in OneRegionTimeseriesDataset. I changed the pattern of the returned values because it seemed like the easiest way to not break existing tests. Users of this dict will need to handle missing data and it doesn't seem horrible for callers to treat the dict element missing the same as the dict containing a None or nan value.

Reply:
I think the only real danger is that somehow a dataset you're using never has a column in it. 

For instance if you dropped a column from the combined datasets map, it wouldn't be immediately obvious that it was actually being used in downstream code because it's obscured by a `.get(` that doesn't fail loudly.

But happy to move in that direction if it makes life easier in refactoring!

Reply:
Yeah, I *really* want a solution that gives us easier visibility into the downstream data changes caused by local code changes :-/